src/site/SiteNavigation.js

```diff
@@ -0,0 +1,207 @@
+
+  if (isSiteNavButtonClick) {
+    handleSiteNavigationButtonClick(clickEvent);
+  } else if (isDisabledPageClick) {
+    // Close the site nav
+    toggleSiteNav(false);
+  }
+
+  return;
+};
+
+/**
+ * Handle any keydown on the document. An switch statement will handle any specific
+ * click events we want to capture.
+ * @param {document:keydown} keyDownEvent - the observable keydown event
+ * @listens document:keydown
+ */
+const handleDocumentKeyDown = keyDownEvent => {
+  switch (event.key) {
+    case "Escape":
+      // Close the site nav
+      toggleSiteNav(false);
+      break;
+    default:
+      return; // Quit when this doesn't handle the key event.
+  }
+
+  // Cancel the default action to avoid it being handled twice
+  event.preventDefault();
+};
+
+/**
+ * Handle the site navigation button click. Works as a toggle button to show / hide site navigation.
+ * @param {button:click} clickEvent - the observable click event
+ * @listens button:click
+ */
+const handleSiteNavigationButtonClick = clickEvent => {
+  const { target } = clickEvent;
+  const siteNavToggleButton =
+    target.id === ids.siteNavToggleButton
+      ? target
+      : target.closest(`#${ids.siteNavToggleButton}`);
+  const buttonAriaExpandedValue = siteNavToggleButton.getAttribute(
+    attributes.ariaExpanded
+  );
+
+  /** If no aria attribute exists this means the menu is closed and we want to show it. */
+  const shouldShowNav =
+    !buttonAriaExpandedValue ||
+    !(buttonAriaExpandedValue.toLowerCase() === "true");
+
+  toggleSiteNav(shouldShowNav);
+};
+
+/**
+ * Do stuff after the dom has loaded
+ */
+const onDocumentReady = () => {
+  // Allows users to use the menu even if javascript is not enabled
+  document
+    .getElementById(ids.siteNavToggleButton)
+    .classList.remove(cssClasses.hiddenInit); //TODO: do we still need this?
+
+  const siteNav = document.getElementById(ids.siteNav);
+
+  document.getElementById(ids.siteNavLinks).classList.add(cssClasses.hidden);
+
+  SetAttribute(
+    siteNav.querySelectorAll("a"),
+    attributes.tabIndex,
+    invisibleTabIndex
+  );
+
+  /**
+   * Initialize a focus trap for use when the site navigation is active / visible
+   * See http://davidtheclark.github.io/focus-trap/ for more details.
+   */
+
+  siteNavFocusTrap = FocusTrap(document.getElementById(ids.siteNav), {
+    clickOutsideDeactivates: true
+  });
+};
+
+/**
+ * Remove site nav items from the tab index if the site nav is hidden
+ * @param {boolean} shouldShow flag that denotes if the site nav should be shown
+ */
+const updateSiteNavTabIndex = shouldShow => {
+  const siteNavLinkElms = document
+    .getElementById(ids.siteNav)
+    .querySelectorAll("a");
+  siteNavLinkElms.forEach(siteNavLinkElm => {
+    if (shouldShow) {
+      siteNavLinkElm.removeAttribute(attributes.tabIndex);
+    } else {
+      siteNavLinkElm.setAttribute(attributes.tabIndex, invisibleTabIndex);
+    }
```

Comment:
Similiar to previous question. You have SetAttribute in your utilities import but arent using it here.

Reply:
I see this is a single element so the utility wouldnt really make sense here either like the above statement.